src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -307,5 +318,35 @@ private static bool SupportsCSharpCodeActions(JToken token)
 
             return providesCodeActions && resolvesCodeActions;
         }
+
+        public override Task<JObject[]> UpdateConfigurationAsync(
+            OmniSharp.Extensions.LanguageServer.Protocol.Models.ConfigurationParams configParams,
+            CancellationToken cancellationToken)
+        {
+            var insertSpaces = _clientOptionsMonitor.InsertSpaces;
+            var tabSize = _clientOptionsMonitor.TabSize;
+
+            var result = new JObject[configParams.Items.Count()];
+            var index = 0;
+            foreach (var item in configParams.Items)
+            {
+                // Right now in VS we only care about editor settings, but we should update this logic later if
+                // we want to support Razor and HTML settings as well.
+                result[index] = item.Section == "editor"
+                    ? JObject.FromObject(new JsonEditorSettings { InsertSpaces = insertSpaces, TabSize = tabSize })
+                    : new JObject();
+
+                index++;
+            }
+
+            return Task.FromResult(result);
+        }
+
+        private struct JsonEditorSettings
```

Comment:
struct should technically be immutable below (there's setters); however, it doesn't buy you much here. For simplicity you could even use records 😄
```C#
private record EditorSettings(bool InsertSpaces, bool TabSize);
```

Orrrr if you want you could just have the lsp options monitor utilize the pre-existing [`EditorSettings`](https://github.com/dotnet/aspnetcore-tooling/blob/main/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Editor/EditorSettings.cs) type and then translate to LSP specific concepts on the server side (insert spaces instead of indentwithtabs). That way you wouldn't have to create an intermediary object and you'd have a single representation of these types of options instead of using tuples/structs in places